### PR TITLE
feat(streaming): spin loop when too many row-id requests in single ms

### PR DESCRIPTION
**Pull request by yezizp2012**
## What's changed and what's your intention?

As title, after https://github.com/singularity-data/risingwave/pull/2205.

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]

File: src/source/src/row_id.rs [hunk 1]
```diff
@@ -68,14 +68,20 @@ impl RowIdGenerator {
...
+                | (self.worker_id << WORKER_ID_SHIFT_BITS) as i64
+                | self.sequence as i64;
+            self.sequence += 1;
 
-        row_id
+            row_id
+        } else {
+            // If the sequence reaches the upper bound, spin loop here and wait for next
+            // millisecond.
+            std::thread::sleep(std::time::Duration::from_millis(1));
```
Comment (fuyufjh) [inline src/source/src/row_id.rs:82]: May print a warning here?

Comment (BugenZhao) [inline src/source/src/row_id.rs:82]: 1ms seems too long? 🤣

Comment (yezizp2012) [reply src/source/src/row_id.rs:83]: We are using ms as timestamp in row-id. Seems okay?

Comment (BugenZhao) [reply src/source/src/row_id.rs:83]: For example when the sequence runs out at the tick of 1.9ms, then we only need to wait for 100us for next sequence number.

Comment (yezizp2012) [reply src/source/src/row_id.rs]: Changed to sleep exact time interval. 😊

**Review by BugenZhao — APPROVED** [review]
LGTM!

**Comment by skyzh** [comment]
In fact, I didn't think this really solves the problem. The clock can go forward or backward after the spin loop. So the next time we acquire an epoch, it might still be less than last generated one.

**Comment by skyzh** [comment]
Okay, maybe this PR doesn't take clock-go-backward into account. Then let's fix this altogether after monolithic clock has been implemented.

**Comment by yezizp2012** [comment]
> Okay, maybe this PR doesn't take clock-go-backward into account. Then let's fix this altogether after monolithic clock has been implemented.

This is just for row-id generate. Yes, clock-go-backward also need to be considered when compute node reboot. This will be fixed together with epoch.

**Pull request merged**